File: 380-insert-delete-getrandom-o1/insert-delete-getrandom-o1.py
class RandomizedSet(object):

    def __init__(self):
        self.SET = set()

    def insert(self, val):
        if val not in self.SET:
            self.SET.add(val)
            return True
        return False
        

    def remove(self, val):
        if val in self.SET:
            self.SET.remove(val)
            return True
        return False
        

    def getRandom(self):
        return random.choice(list(self.SET))



# Your RandomizedSet object will be instantiated and called as such:
# obj = RandomizedSet()
# param_1 = obj.insert(val)
# param_2 = obj.remove(val)
# param_3 = obj.getRandom()

File: 380-insert-delete-getrandom-o1/insert-delete-getrandom-o1.java
class RandomizedSet {

    HashSet<Integer> set;
    ArrayList<Integer> list;

    public RandomizedSet() {
        set = new HashSet<>();
        list = new ArrayList<>();
    }
    
    public boolean insert(int val) {

        if (!set.contains(val)) {
            set.add(val);
            list.add(val);
            return true;
        }
        return false;
    }
    
    public boolean remove(int val) {

        if (set.contains(val)) {
            set.remove(val);
            list.remove(Integer.valueOf(val));
            return true;
        }
        return false;
    }
    
    public int getRandom() {

        int index = (int)(Math.random() * list.size());
        return list.get(index);
    }
}